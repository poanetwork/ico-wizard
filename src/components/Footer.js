import React from 'react'
import '../assets/stylesheets/application.css';
import { Link } from 'react-router-dom'

export const Footer = () => (
	<footer className="footer">
		<div className="container">
			<p className="rights">2017 Oracles. All rights reserved.</p>
      <Link className="logo" to='/'></Link>
			<div className="socials">
				<button className="social social_reddit"></button>
				<button className="social social_twitter"></button>
				<button className="social social_oracles"></button>
				<button className="social social_telegram"></button>
				<button className="social social_github"></button>
			</div>
		</div>
	</footer>
)