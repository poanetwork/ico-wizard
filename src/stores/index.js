import contractStore from './ContractStore';
import pricingStrategyStore from './PricingStrategyStore';
import reservedTokenStore from './ReservedTokenStore';
import reservedTokenInputStore from './ReservedTokenInputStore';
import reservedTokenElementStore from './ReservedTokenElementStore';
import stepThreeValidationStore from './StepThreeValidationStore';
import stepTwoValidationStore from './StepTwoValidationStore';
import tierStore from './TierStore';
import tokenStore from './TokenStore';
import web3Store from './Web3Store';
import tierCrowdsaleListStore from './TierCrowdsaleListStore'
import crowdsaleBlockListStore from './CrowdsaleBlockListStore'

export {
    tierCrowdsaleListStore,
    crowdsaleBlockListStore,
    contractStore,
    pricingStrategyStore,
    reservedTokenStore,
    reservedTokenInputStore,
    reservedTokenElementStore,
    stepThreeValidationStore,
    stepTwoValidationStore,
    tierStore,
    tokenStore,
    web3Store
};
