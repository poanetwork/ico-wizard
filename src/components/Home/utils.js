import { DOWNLOAD_STATUS } from '../../utils/constants'
import { getNetworkVersion } from '../../utils/blockchainHelpers'
import { getCrowdsaleAssets } from '../../stores/utils'
import logdown from 'logdown'

const logger = logdown('TW:home:utils')

export const reloadStorage = async props => {
  let { generalStore, contractStore } = props

  try {
    if (!contractStore || !generalStore) {
      throw new Error('There is no stores to set')
    }
    contractStore.setProperty('downloadStatus', DOWNLOAD_STATUS.PENDING)

    // General store, check network
    let networkID = await getNetworkVersion()
    generalStore.setProperty('networkID', networkID)

    // Contract store, get contract and abi
    await getCrowdsaleAssets(networkID)
    contractStore.setProperty('downloadStatus', DOWNLOAD_STATUS.SUCCESS)
    return true
  } catch (e) {
    logger.error('Error downloading contracts', e)
    if (contractStore) {
      contractStore.setProperty('downloadStatus', DOWNLOAD_STATUS.FAILURE)
    }
    throw e
  }
}
