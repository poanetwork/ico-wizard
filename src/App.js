import React, {Component} from 'react';
import './assets/stylesheets/application.css';
import { Header, Footer, Home, stepOne, stepTwo, stepThree, stepFour, stepFive, Invest } from './components/index'
import { getQueryVariable } from './utils/utils'
import {
  BrowserRouter as Router,
  Route
} from 'react-router-dom'

class App extends Component {
  render() {
    var crowdsaleAddr = getQueryVariable("addr");
    return (
      <Router>
        <div>
          <Header/>
            <Route exact path="/" component={crowdsaleAddr?stepFive:Home}/>
            <Route path="/1" component={stepOne}/>
            <Route path="/2" component={stepTwo}/>
            <Route path="/3" component={stepThree}/>
            <Route path="/4" component={stepFour}/>
            <Route path="/crowdsale" component={stepFive}/>
            <Route path="/invest" component={Invest}/>
          <Footer/>
        </div>
      </Router>
    )
  }
}

export default App;
