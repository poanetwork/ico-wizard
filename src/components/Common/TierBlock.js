import React from 'react'
import { Field } from 'react-final-form'
import { InputField2 } from './InputField2'
import { WhitelistInputBlock } from './WhitelistInputBlock'
import { composeValidators, isGreaterOrEqualThan, isMaxLength, isPositive, isRequired } from '../../utils/validations'
import { DESCRIPTION, TEXT_FIELDS } from '../../utils/constants'
import { CrowdsaleStartTime } from './CrowdsaleStartTime'
import { CrowdsaleEndTime } from './CrowdsaleEndTime'
import { CrowdsaleRate } from './CrowdsaleRate'
import { MinCap } from './MinCap'
import { acceptPositiveIntegerOnly } from '../../utils/utils'

const { ALLOW_MODIFYING, CROWDSALE_SETUP_NAME, ENABLE_WHITELISTING } = TEXT_FIELDS

const inputErrorStyle = {
  color: 'red',
  fontWeight: 'bold',
  fontSize: '12px',
  width: '100%',
  height: '20px'
}

export const TierBlock = ({ fields, ...props }) => {
  return (
    <div>
      {fields.map((name, index) => (
        <div style={{ marginTop: '40px' }} className="steps-content container" key={index}>
          <div className="hidden">
            <div className="input-block-container">
              <Field
                id={`${name}.tier`}
                name={`${name}.tier`}
                validate={value => {
                  const errors = composeValidators(isRequired(), isMaxLength()(30))(value)

                  if (errors) return errors.shift()
                }}
                errorStyle={inputErrorStyle}
                component={InputField2}
                type="text"
                side="left"
                label={CROWDSALE_SETUP_NAME}
                description={DESCRIPTION.CROWDSALE_SETUP_NAME}
              />
            </div>

            <div className="input-block-container">
              <Field
                id={`${name}.updatable`}
                name={`${name}.updatable`}
                render={({ input }) => (
                  <div className="left">
                    <label className="label">{ALLOW_MODIFYING}</label>
                    <div className="radios-inline">
                      <label className="radio-inline">
                        <input
                          id={`${name}.allow_modifying_on`}
                          type="radio"
                          checked={input.value === 'on'}
                          onChange={() => input.onChange('on')}
                          value="on"
                        />
                        <span className="title">on</span>
                      </label>
                      <label className="radio-inline">
                        <input
                          id={`${name}.allow_modifying_off`}
                          type="radio"
                          checked={input.value === 'off'}
                          value="off"
                          onChange={() => input.onChange('off')}
                        />
                        <span className="title">off</span>
                      </label>
                    </div>
                    <p className="description">{DESCRIPTION.ALLOW_MODIFYING}</p>
                  </div>
                )}
              />

              <Field
                name={`${name}.whitelistEnabled`}
                render={({ input }) => (
                  <div className="right">
                    <label className="label">{ENABLE_WHITELISTING}</label>
                    <div className="radios-inline">
                      <label className="radio-inline">
                        <input
                          id={`${name}.enable_whitelisting_yes`}
                          type="radio"
                          checked={input.value === 'yes'}
                          value="yes"
                          onChange={() => input.onChange('yes')}
                        />
                        <span className="title">yes</span>
                      </label>
                      <label className="radio-inline">
                        <input
                          id={`${name}.enable_whitelisting_no`}
                          type="radio"
                          checked={input.value === 'no'}
                          value="no"
                          onChange={() => input.onChange('no')}
                        />
                        <span className="title">no</span>
                      </label>
                    </div>
                    <p className="description">{DESCRIPTION.ENABLE_WHITELIST}</p>
                  </div>
                )}
              />
            </div>

            <div className="input-block-container">
              <CrowdsaleStartTime
                name={`${name}.startTime`}
                index={index}
                disabled={index > 0}
                side="left"
                errorStyle={inputErrorStyle}
              />
              <CrowdsaleEndTime name={`${name}.endTime`} index={index} side="right" errorStyle={inputErrorStyle} />
            </div>

            <div className="input-block-container">
              <CrowdsaleRate name={`${name}.rate`} errorStyle={inputErrorStyle} side="left" />
              <Field
                name={`${name}.supply`}
                component={InputField2}
                validate={(value, values) => {
                  const { tierStore } = props
                  const listOfValidations = [isPositive()]

                  if (values.tiers[index].whitelistEnabled === 'yes') {
                    const maxCapSum = tierStore.whitelistMaxCapSum[index]
                    listOfValidations.push(
                      isGreaterOrEqualThan(`Can't be less than sum of whitelist's maxCap: ${maxCapSum}`)(maxCapSum)
                    )
                  }

                  const errors = composeValidators(...listOfValidations)(value)
                  if (errors) return errors.shift()
                }}
                parse={acceptPositiveIntegerOnly}
                errorStyle={inputErrorStyle}
                type="text"
                side="right"
                label={TEXT_FIELDS.SUPPLY_SHORT}
                description={DESCRIPTION.SUPPLY}
              />
            </div>
            <div className="input-block-container">
              <MinCap
                name={`${name}.minCap`}
                errorStyle={inputErrorStyle}
                decimals={props.decimals}
                index={index}
                disabled={props.tierStore ? props.tierStore.tiers[index].whitelistEnabled === 'yes' : true}
                side="left"
              />
            </div>
          </div>
          {props.tierStore.tiers[index].whitelistEnabled === 'yes' ? (
            <div>
              <div className="section-title">
                <p className="title">Whitelist</p>
              </div>
              <WhitelistInputBlock num={index} decimals={props.decimals} supply={props.tierStore.tiers[index].supply} />
            </div>
          ) : null}
        </div>
      ))}
    </div>
  )
}
