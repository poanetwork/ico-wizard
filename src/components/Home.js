import React, { Component } from 'react';
import '../assets/stylesheets/application.css';
import { Link } from 'react-router-dom'
import { setFlatFileContentToState } from '../utils/utils';
import { defaultState } from '../utils/constants'

export class Home extends Component {
  constructor(props) {
    super(props);
    this.state = defaultState
  }

  addContractsToState (src, bin, abi) {
    let newState = Object.assign({}, this.state)
    newState.contracts.crowdsale = {
      src,
      bin,
      abi: JSON.parse(abi)
    }
    this.setState(newState)
  }

  componentDidMount() {
    const contractName = "SampleCrowdsale";
    let src, bin
    //var contractName = "RomanCrowdsale";
    setFlatFileContentToState("./contracts/" + contractName + "_flat.sol", (content) => src = content);
    setFlatFileContentToState("./contracts/" + contractName + "_flat.bin", (_bin) => bin = _bin);
    setFlatFileContentToState("./contracts/" + contractName + "_flat.abi", (_abi) => this.addContractsToState(src, bin, _abi));
    /*setFlatFileContentToState("./contracts/SampleCrowdsaleToken_flat.bin", function(content) {
      $this.state.contracts.token.bin = content;
    });
    setFlatFileContentToState("./contracts/SampleCrowdsaleToken_flat.abi", function(content) {
      $this.state.contracts.token.abi = JSON.parse(content);
    });*/
  }

  render() {
    return (
      <div>
        <section className="home">
          <div className="crowdsale">
            <div className="container">
              <h1 className="title">Create crowdsale</h1>
              <p className="description">
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
                veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea  commodo consequat.
              </p>
              <div className="buttons">
                <Link to={{ pathname: '/1', query: { state: this.state } }}><a className="button button_fill">New crowdsale</a></Link>
              </div>
            </div>
          </div>
          <div className="process">
            <div className="container">
              <div className="process-item">
                <div className="step-icons step-icons_crowdsale-contract"></div>
                <p className="title">Crowdsale Contract</p>
                <p className="description">
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
                </p>
              </div>
              <div className="process-item">
                <div className="step-icons step-icons_token-setup"></div>
                <p className="title">Token Setup</p>
                <p className="description">
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
                </p>
              </div>
              <div className="process-item">
                <div className="step-icons step-icons_crowdsale-setup"></div>
                <p className="title">Crowdsale Setup</p>
                <p className="description">
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
                </p>
              </div>
              <div className="process-item">
                <div className="step-icons step-icons_publish"></div>
                <p className="title">Publish</p>
                <p className="description">
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
                </p>
              </div>
              <div className="process-item">
                <div className="step-icons step-icons_crowdsale-page"></div>
                <p className="title">Crowdsale Page</p>
                <p className="description">
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
                </p>
              </div>
            </div>
          </div>
        </section>
      </div>
    );
  }
}