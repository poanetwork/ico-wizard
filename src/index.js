import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import registerServiceWorker from './registerServiceWorker';
import { useStrict } from 'mobx';
import { Provider } from 'mobx-react';
import * as stores from './stores';
useStrict(true);


ReactDOM.render(
  <Provider { ...stores }>
    <App />
  </Provider>, 
document.getElementById('root'));
registerServiceWorker();
