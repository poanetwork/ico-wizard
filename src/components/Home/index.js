import React, { Component } from 'react';
import '../../assets/stylesheets/application.css';
import { Link } from 'react-router-dom'
import CrowdsalesList from '../Common/CrowdsalesList'
import { Loader } from '../Common/Loader'
import { loadRegistryAddresses } from '../../utils/blockchainHelpers'

export class Home extends Component {
  constructor (props) {
    super(props)
    this.state = {
      showModal: false,
      loading: false
    }
  }

  chooseContract = () => {
    this.setState({
      loading: true
    })

    loadRegistryAddresses()
      .then(() => {
        this.setState({
          loading: false,
          showModal: true
        })
      }, (e) => {
        console.error('There was a problem loading the crowdsale addresses from the registry', e)
        this.setState({
          loading: false
        })
      })
  }

  onClick = crowdsaleAddress => {
    this.props.history.push('/manage/' + crowdsaleAddress)
  }

  renderModal = () => {
    return (
      <div className="crowdsale-modal loading-container">
        <div className='modal'>
          <p className='title'>Crowdsales List</p>
          <p className='description'>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Consectetur cumque dolore
            doloribus ea earumeligendi eveniet ex expedita itaque laborum maxime mollitia nemo nihil perferendis
            quibusdam sed sequiullam, vitae!</p>
          <CrowdsalesList onClick={this.onClick}/>
          <div className='close-button' onClick={() => this.hideModal()}>X</div>
        </div>
      </div>
    )
  }

  hideModal = () => {
    this.setState({ showModal: false })
  }

  render () {
    const chooseContract = process.env.NODE_ENV === 'development'
      ? <div onClick={() => this.chooseContract()} className="button button_outline">Choose Contract</div>
      : null

    const chooseContractModal = this.state.showModal
      ? this.renderModal()
      : null

    return (
      <div>
        <section className="home">
          <div className="crowdsale">
            <div className="container">
              <h1 className="title">Welcome to ICO Wizard</h1>
              <p className="description">
              ICO Wizard is a client side tool to create token and crowdsale contracts in five steps. It helps you to publish contracts on the Ethereum network, verify them in Etherscan, create a crowdsale page with stats. For participants, the wizard creates a page to invest into the campaign.
              <br/>Smart contracts based on <a href="https://github.com/TokenMarketNet/ico">TokenMarket</a> contracts.
              </p>
              <div className="buttons">
                <Link to='/1'><span className="button button_fill">New crowdsale</span></Link>
                {chooseContract}
              </div>
            </div>
          </div>
          <div className="process">
            <div className="container">
              <div className="process-item">
                <div className="step-icons step-icons_crowdsale-contract"></div>
                <p className="title">Crowdsale Contract</p>
                <p className="description">
                  Select a strategy for crowdsale contract
                </p>
              </div>
              <div className="process-item">
                <div className="step-icons step-icons_token-setup"></div>
                <p className="title">Token Setup</p>
                <p className="description">
                  Setup token and reserved distribution
                </p>
              </div>
              <div className="process-item">
                <div className="step-icons step-icons_crowdsale-setup"></div>
                <p className="title">Crowdsale Setup</p>
                <p className="description">
                  Setup tiers and crowdsale parameters
                </p>
              </div>
              <div className="process-item">
                <div className="step-icons step-icons_publish"></div>
                <p className="title">Publish</p>
                <p className="description">
                  Get generated code and artifacts for verification in Etherscan
                </p>
              </div>
              <div className="process-item">
                <div className="step-icons step-icons_crowdsale-page"></div>
                <p className="title">Crowdsale Page</p>
                <p className="description">
                  Bookmark this page for the campaign statistics
                </p>
              </div>
            </div>
          </div>
          {chooseContractModal}
          <Loader show={this.state.loading}></Loader>
        </section>
      </div>
    );
  }
}
