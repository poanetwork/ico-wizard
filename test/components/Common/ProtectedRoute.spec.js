import React from 'react'
import render from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import { configure, mount, shallow } from 'enzyme'
import { ProtectedRoute } from '../../../src/components/Common/ProtectedRoute'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import { deploymentStore } from '../../../src/stores'
import { Home } from '../../../src/components/Home'
import { Provider } from 'mobx-react'

configure({ adapter: new Adapter() })

describe(`ProtectedRoute`, () => {
  it(`should render ProtectedRoute component`, () => {
    const stores = { deploymentStore }

    const wrapper = mount(
      <Provider {...stores}>
        <Router>
          <ProtectedRoute {...stores} component={Home} />
        </Router>
      </Provider>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
