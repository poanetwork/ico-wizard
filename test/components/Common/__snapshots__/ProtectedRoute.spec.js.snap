// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtectedRoute should render ProtectedRoute component 1`] = `
<Provider
  deploymentStore={
    DeploymentStore {
      "deployerAccount": null,
      "deploymentStep": null,
      "hasEnded": false,
      "invalidAccount": false,
      "logTxMap": [Function],
      "txMap": Object {},
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <inject-ProtectedRoute-with-deploymentStore
        component={[Function]}
        deploymentStore={
          DeploymentStore {
            "deployerAccount": null,
            "deploymentStep": null,
            "hasEnded": false,
            "invalidAccount": false,
            "logTxMap": [Function],
            "txMap": Object {},
          }
        }
      >
        <ProtectedRoute
          component={[Function]}
          deploymentStore={
            DeploymentStore {
              "deployerAccount": null,
              "deploymentStep": null,
              "hasEnded": false,
              "invalidAccount": false,
              "logTxMap": [Function],
              "txMap": Object {},
            }
          }
        >
          <Route
            render={[Function]}
          >
            <inject-Home-with-deploymentStore
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <Home
                deploymentStore={
                  DeploymentStore {
                    "deployerAccount": null,
                    "deploymentStep": null,
                    "hasEnded": false,
                    "invalidAccount": false,
                    "logTxMap": [Function],
                    "txMap": Object {},
                  }
                }
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
              >
                <div>
                  <section
                    className="hm-Home"
                  >
                    <div
                      className="hm-Home_Scroll"
                    >
                      <div
                        className="hm-Home_Top"
                      >
                        <LogoPrimary>
                          <Link
                            className="sw-LogoPrimary"
                            replace={false}
                            to="/"
                          >
                            <a
                              className="sw-LogoPrimary"
                              href="/"
                              onClick={[Function]}
                            />
                          </Link>
                        </LogoPrimary>
                      </div>
                      <div
                        className="hm-Home_MainInfoContainer"
                      >
                        <div
                          className="hm-Home_MainInfo"
                        >
                          <div
                            className="hm-Home_MainInfoTextContainer"
                          >
                            <h1
                              className="hm-Home_MainInfoTitle"
                            >
                              Welcome to Token Wizard
                            </h1>
                            <p
                              className="hm-Home_MainInfoDescription"
                            >
                              Token Wizard is a client side tool to create ERC20 token and crowdsale in five steps. It helps you to publish contracts on the Ethereum network, create a crowdsale page with stats. For participants, the wizard creates a page to contribute into the campaign.
                            </p>
                            <p
                              className="hm-Home_MainInfoPoweredBy"
                            >
                              Token Wizard is powered by 
                              <a
                                href="https://github.com/auth-os/beta"
                              >
                                Auth-os
                              </a>
                              .
                            </p>
                          </div>
                          <div
                            className="hm-Home_MainInfoButtonContainer"
                          >
                            <ButtonCreateCrowdsale
                              history={
                                Object {
                                  "action": "POP",
                                  "block": [Function],
                                  "createHref": [Function],
                                  "go": [Function],
                                  "goBack": [Function],
                                  "goForward": [Function],
                                  "length": 1,
                                  "listen": [Function],
                                  "location": Object {
                                    "hash": "",
                                    "pathname": "/",
                                    "search": "",
                                    "state": undefined,
                                  },
                                  "push": [Function],
                                  "replace": [Function],
                                }
                              }
                            >
                              <button
                                className="hm-ButtonCreateCrowdsale"
                                onClick={[Function]}
                              >
                                New crowdsale
                              </button>
                            </ButtonCreateCrowdsale>
                            <ButtonChooseCrowdsale
                              history={
                                Object {
                                  "action": "POP",
                                  "block": [Function],
                                  "createHref": [Function],
                                  "go": [Function],
                                  "goBack": [Function],
                                  "goForward": [Function],
                                  "length": 1,
                                  "listen": [Function],
                                  "location": Object {
                                    "hash": "",
                                    "pathname": "/",
                                    "search": "",
                                    "state": undefined,
                                  },
                                  "push": [Function],
                                  "replace": [Function],
                                }
                              }
                            >
                              <div
                                className="hm-ButtonChooseCrowdsale"
                                onClick={[Function]}
                              >
                                Choose Crowdsale
                              </div>
                            </ButtonChooseCrowdsale>
                          </div>
                        </div>
                      </div>
                    </div>
                  </section>
                </div>
              </Home>
            </inject-Home-with-deploymentStore>
          </Route>
        </ProtectedRoute>
      </inject-ProtectedRoute-with-deploymentStore>
    </Router>
  </BrowserRouter>
</Provider>
`;
